frontend/src/routes/Governance/discovered/useFetchPolicies.tsx

```diff
@@ -0,0 +1,111 @@
+/* Copyright Contributors to the Open Cluster Management project */
+// This react hook will be tested in e2e test(Cypress) due to its use of workers and blobs.
+import { useEffect, useState } from 'react'
+import { grouping } from './grouping'
+import { useRecoilValue, useSharedAtoms } from '../../../shared-recoil'
+import { useSearchResultItemsQuery } from '../../Search/search-sdk/search-sdk'
+import { searchClient } from '../../Search/search-sdk/search-client'
+import { convertStringToQuery } from '../../Search/search-helper'
+import { parseDiscoveryPolicies } from '../common/util'
+export interface ISourceType {
+  type: string //ex: 'Policy' | 'Git' | 'Multiple'
+  parentNs: string
+  parentName: string
+}
+export interface DiscoverdPolicyItem {
+  _hubClusterResource: boolean
+  kind: string
+  apigroup: string
+  apiversion: string
+  cluster: string
+  compliant: string
+  remediationAction: string
+  severity: string
+  _isExternal: boolean
+  annotations: { [annoKey: string]: string }
+  _uid: string
+  created: string
+  label: string
+  kind_plural: string
+  namespace: string
+  name: string
+  disabled: boolean
+  // These are only for Operator policy
+  deploymentAvailable?: boolean | 'yes' | 'no'
+  upgradeAvailable?: boolean | 'yes' | 'no'
+  // Not from search-collector. Attached in grouping function
+  source?: ISourceType
+}
+
+export interface DiscoverdPolicyTableItem {
+  id: string
+  name: string
+  severity: string
+  kind: string
+  responseAction: string
+  policies: DiscoverdPolicyItem[]
+  source?: ISourceType
+}
+
+// If id (`policyName` + `policyKind`) exists, it returns a filtered `DiscoveredPolicyTable` based on `clusterName`.
+export function useFetchPolicies(id?: string) {
+  const [isFetching, setIsFetching] = useState(true)
+  const [data, setData] = useState<DiscoverdPolicyTableItem[]>()
+  const { channelsState, helmReleaseState, subscriptionsState } = useSharedAtoms()
+  const helmReleases = useRecoilValue(helmReleaseState)
+  const subscriptions = useRecoilValue(subscriptionsState)
+  const channels = useRecoilValue(channelsState)
+  const {
+    data: searchData,
+    loading: searchLoading,
+    error: searchErr,
+  } = useSearchResultItemsQuery({
+    client: process.env.NODE_ENV === 'test' ? undefined : searchClient,
+    variables: { input: [convertStringToQuery('kind:ConfigurationPolicy,CertificatePolicy,OperatorPolicy', 100000)] },
```

Comment:
This filter is fine for the initial "Discovered policies" table but when `id` is specified, it needs to target the specific policy kind, apiVersion/group, and policy name. `id` should likely be multiple arguments instead.